### ISSUE TITLE
Too many IKE transforms in ike-version.nse

### ISSUE THREAD

**Issue by nnposter** [nmap/nmap#524]
Script `ike-version.nse` is sending IKE packets with a large number of transforms, 120 in total, by proposing every possible combination of:
* 4 authentication types
* 5 symmetric ciphers
* 2 hashes
* 3 DH groups

The net result is a UDP packet with IKE payload of 4,664 bytes in main mode or 4,828 bytes in aggressive mode. This is not desirable as the packet gets fragmented, which in turn causes problems with packet filters and stateful firewalls. (As an example, see a [Cisco note](https://www.cisco.com/c/en/us/td/docs/ios-xml/ios/sec_conn_ikevpn/configuration/15-mt/sec-ike-for-ipsec-vpns-15-mt-book/sec-fragment-ike-pack.html) on this issue.)

The script probably needs to be changed to use only specifically chosen transforms per packet, keeping the total count below about 30.

**Comment by dmiller-nmap** [comment]
It seems we could limit these to reasonable combinations or even send more than one packet per mode. We've had success with other trial-and-error handshake protocols like TLS in breaking up possibilities into reasonable-sized chunks. I would be interested to learn more about the protocol and whether the server responds with only one chosen transform or all supported ones, whether the server can support more than one transform, and other questions of that sort.

**Comment by pandread** [comment]
Yes usually the server has his own selection of available transforms (cipher suites). The client sends what he can support and the server chooses with regards to his transforms and probably the optimal one. 

Splitting the payload into more than one packet is the only option. There are 2 ways:
- First one would be assigning a value of maximum number of transforms in a packet and loading up transforms with regards to maximum non-fragmented packet size.

- Another way would be creating a packet for each transform...Would the second option imply a DoS and should be avoided? 

I am currently exploring [](https://github.com/nmap/nmap/blob/master/nselib/ike.lua) which seems to be the main target for modifications here. I think the functions to create the transforms are kinda confusing and could be simpler. There could definitely exist a `struct transforms` of certain size (size as much as it fits for a packet to not get fragmented). So we load up structs like that in a while loop and feed them to separate packets.  
Feeding into packets might also require modifications of code. A loop and a counter could do the trick. 

I am interested if someone has searched deeper and knows what is going on. I am interested more specifically on the IKE-enum script `ike-scan` and would like to know if there is someone assigned to it, but working on ike.lua will be surely required for both.
